src/web/js/trove/chart-lib.js

```diff
@@ -1601,6 +1622,43 @@
               svgRoot.appendChild(imageElt);
             });
           });
+          }
+
+          if (dotChartP) {
+            const circles = [...svgRoot.children[1].children[2].getElementsByTagName('circle')];
+            // console.log('circles=', circles);
+            const numCircles = circles.length;
+            const circle0 = circles[0];
+            const circleR = toFixnum(get(combined[0], 'point-size'));
+            // const circleR = Number(circle0.getAttribute('r'));
+            const offsetQuantum = 2 * circleR;
+            let prevDotArray = [];
+            function tooClose(x, y) {
+              return prevDotArray.some(function(n) {
+                return ((Math.abs(x - n[0]) < offsetQuantum) &&
+                  (Math.abs(y - n[1]) < offsetQuantum));
+              });
+            }
+
+            circles.forEach((circle) => {
+              // console.log('updating circle', i);
+              const circleX = Number(circle.getAttribute('cx'));
+              let circleY = Number(circle.getAttribute('cy')) - 1.2*circleR;
+              while (tooClose(circleX, circleY)) {
+                circleY -= offsetQuantum;
+              }
+
+              prevDotArray.push([circleX, circleY]);
+
+              const circleElt = circle.cloneNode(false);
+              circleElt.classList.add('__img_labels'); // tag for later gc
+              circleElt.setAttribute('cy', circleY);
+              circleElt.setAttribute('r', circleR);
+              circleElt.setAttribute('fill-opacity', 1);
```

Comment:
We can drop `setAttribute` for `r` and `fill-opacity`.

Reply:
These seem to be needed. The circles/dots aren't drawn otherwise

Reply:
Huh. Even if you use `cloneNode(true)`?

Reply:
Ah, of course -- because the circles we're cloning are already invisible. :P 

Nevermind. My bad.